import "ydb/public/api/protos/ydb_export.proto";
import "ydb/public/api/protos/ydb_issue_message.proto";
import "ydb/public/api/protos/ydb_operation.proto";
import "ydb/public/api/protos/ydb_status_codes.proto";

import "google/protobuf/timestamp.proto";

package NKikimrExport;
option java_package = "ru.yandex.kikimr.proto";

message TExport {
    optional uint64 Id = 1;
    optional Ydb.StatusIds.StatusCode Status = 2;
    repeated Ydb.Issue.IssueMessage Issues = 3;
    optional Ydb.Export.ExportProgress.Progress Progress = 4;
    optional google.protobuf.Timestamp StartTime = 8;
    optional google.protobuf.Timestamp EndTime = 9;
    repeated Ydb.Export.ExportItemProgress ItemsProgress = 7;
    oneof Settings {
        Ydb.Export.ExportToYtSettings ExportToYtSettings = 5;
        Ydb.Export.ExportToS3Settings ExportToS3Settings = 6;
    }
}

message TCreateExportRequest {
    optional Ydb.Operations.OperationParams OperationParams = 1;
    oneof Settings {
        Ydb.Export.ExportToYtSettings ExportToYtSettings = 2;
        Ydb.Export.ExportToS3Settings ExportToS3Settings = 3;
    }
}

message TEvCreateExportRequest {
    optional uint64 TxId = 1;
    optional string DatabaseName = 2;
    optional string UserSID = 4;
    optional TCreateExportRequest Request = 3;
}

message TCreateExportResponse {
    optional TExport Entry = 1;
}

message TEvCreateExportResponse {
    optional uint64 TxId = 1;
    optional TCreateExportResponse Response = 2;
}

message TGetExportRequest {
    optional uint64 Id = 1;
}

message TEvGetExportRequest {
    optional string DatabaseName = 2;
    optional TGetExportRequest Request = 1;
}

message TGetExportResponse {
    optional TExport Entry = 1;
}

message TEvGetExportResponse {
    optional TGetExportResponse Response = 1;
}

message TCancelExportRequest {
    optional uint64 Id = 1;
}

message TEvCancelExportRequest {
    optional uint64 TxId = 1;
    optional string DatabaseName = 3;
    optional TCancelExportRequest Request = 2;
}

message TCancelExportResponse {
    optional Ydb.StatusIds.StatusCode Status = 1;
    repeated Ydb.Issue.IssueMessage Issues = 2;
}

message TEvCancelExportResponse {
    optional uint64 TxId = 1;
    optional TCancelExportResponse Response = 2;
}

message TForgetExportRequest {
    optional uint64 Id = 1;
}

message TEvForgetExportRequest {
    optional uint64 TxId = 1;
    optional string DatabaseName = 3;
    optional TForgetExportRequest Request = 2;
}

message TForgetExportResponse {
    optional Ydb.StatusIds.StatusCode Status = 1;
    repeated Ydb.Issue.IssueMessage Issues = 2;
}

message TEvForgetExportResponse {
    optional uint64 TxId = 1;
    optional TForgetExportResponse Response = 2;
}

message TListExportsRequest {
    optional uint64 PageSize = 1;
    optional string PageToken = 2;
    optional string Kind = 3;
}

message TEvListExportsRequest {
    optional string DatabaseName = 2;
    optional TListExportsRequest Request = 1;
}

message TListExportsResponse {
    optional Ydb.StatusIds.StatusCode Status = 1;
    repeated Ydb.Issue.IssueMessage Issues = 2;
    repeated TExport Entries = 3;
    optional string NextPageToken = 4;
}

message TEvListExportsResponse {
    optional TListExportsResponse Response = 1;
}
